src/app/CASESessionManager.cpp

```diff
@@ -118,10 +128,11 @@ OperationalSessionSetup * CASESessionManager::FindExistingSessionSetup(const Sco
     return mConfig.sessionSetupPool->FindSessionSetup(peerId, forAddressUpdate);
 }
 
-Optional<SessionHandle> CASESessionManager::FindExistingSession(const ScopedNodeId & peerId) const
+Optional<SessionHandle> CASESessionManager::FindExistingSession(const ScopedNodeId & peerId,
+                                                                const bool useTransportForLargePayload) const
```

Comment:
Using a bool makes the callsites pretty inscrutable....

It seems like what we really want here is an enum, probably with these values:

1. Require large payload (TCP sessions only)
2. Allow large payload (TCP or UDP; caller does not care)
3. Disallow large payload (UDP only; maybe this should be characterized in some other way?  Why would people want to ensure they get a UDP session?)

And the public API should probably reflect this set of things...

Reply:
We may have use cases that need UDP only because of latency concerns....

Reply:
That is a fair point. However, we might need to think through how we need to characterize this. If vying for latency, presumably it is vanilla UDP because MRP would not serve latency needs that well. From that standpoint, what I am thinking is the following:
1. ForLargePayload(TCP sessions only)
2. LatencyOptimized(Vanilla UDP only, although there is nothing today to ensure that MRP is not adopted atop). It is up to the app to choose it, right?
3. Default(MRP or TCP) 
AllowLargePayload(TCP or UDP) does not make much sense because if it is either, it may not allow large payload over UDP, i.e.(as it stands today).
We can avoid this complexity for now and just keep it as LargePayload::Yes or No as well, and visit this later if a low latency vanilla-UDP-only-transport is desired and the layers can ensure that.

Reply:
MRP is fine from a latency perspective, in the sense that sending a new MRP message does not wait for anything previously sent.